WordPress/src/main/java/org/wordpress/android/ui/sitecreation/usecases/FetchSegmentsUseCase.kt

```diff
@@ -0,0 +1,39 @@
+package org.wordpress.android.ui.sitecreation.usecases
+
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.VerticalActionBuilder
+import org.wordpress.android.fluxc.store.VerticalStore.OnSegmentsFetched
+import javax.inject.Inject
+import kotlin.coroutines.experimental.Continuation
+import kotlin.coroutines.experimental.suspendCoroutine
+
+/**
+ * Transforms EventBus event to a coroutines.
+ */
+class FetchSegmentsUseCase @Inject constructor(val dispatcher: Dispatcher) {
+    private var continuation: Continuation<OnSegmentsFetched>? = null
+
+    suspend fun fetchCategories(): OnSegmentsFetched {
+        if (continuation != null) {
+            throw IllegalStateException("Fetch already in progress.")
```

Comment:
I always try to avoid having a code which can be ignored as it makes things a bit less transparent. It's a responsibility of the client to make sure the method can be called only when it makes sense to call it -> for example hide the `retry` button as soon as its tapped.

Nevertheless, if you think it's not a good approach I'm open to removing the exception.

Reply:
I think it's not just about whether the action is destructive or not. It's more about hiding an issue in the code which can results in inconsistent state. If we don't expect it and it's a bug when it happens, I think it's still fine to crash the app. But I've added there the `DEBUG` flag as this is really an edge case.